### PR TITLE
RFC Adds swaggerValidatorOptions.responseValidatorHook

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

**Comment by ghost** [comment]
This is the groundwork for https://www.pivotaltracker.com/story/show/132942081

File: middleware/swagger-validator.js [hunk 1]
```diff
@@ -298,7 +298,11 @@ var wrapEnd = function (req, res, next) {
         mHelpers.debugError(err, debug);
       }
 
-      return next(err);
+      if (responseValidatorHook) {
```
Comment (mhfrantz) [inline middleware/swagger-validator.js:301]: How does inserting the hook here help more than just chaining to another Express middleware?  Isn't `next(err)` essentially performing the same function?

Comment (ghost) [reply middleware/swagger-validator.js:301]: `next(err)` will jump to the error handling middleware in app.js, bypassing the rest of the middleware stack. If any.

Comment (mhfrantz) [reply middleware/swagger-validator.js:301]: I read up on Express middleware error handling, and I see that if you want to suppress an error and continue processing the request, you can't call `next(err)`, so I understand why this hook is justified (and how Express is limited).

**Review by mhfrantz — APPROVED** [review]
LGTM

**Comment by mhfrantz** [comment]
Not merging because of RFC status.

**Comment by ghost** [comment]
Thanks. Please see https://github.com/unitive-inc/swagger-tools/pull/2

**Pull request closed without merging**